index.js

```diff
@@ -1,15 +1,16 @@
 // @ts-check
-var react = require('react')
-module.exports = function UseStateRef(defaultValue) {
-	var [state, setState] = react.useState(defaultValue)
-	var ref = react.useRef(defaultValue)
-	ref.current = state
-	return [
-		state,
-		function (newValue) {
-			ref.current = newValue
-			return setState(newValue)
-		},
-		ref
-	]
-}
\ No newline at end of file
+var React = require('react');
+
+module.exports = function useStateRef(defaultValue) {
+  var [state, setState] = React.useState(defaultValue);
+  var ref = React.useRef(defaultValue);
+
+  var dispatch = React.useCallback(function(val) {
+    ref.current = typeof val === "function" ?
+    val(ref.current) : val;
+
+    setState(val);
```

Comment:
I think you have to change it to `setState(ref.current)` if not, when val is a function it will be executed twice (once by this module, and once by react internals setState)